packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -11,149 +11,227 @@ const ruleTester = new RuleTester({
   parser: '@typescript-eslint/parser',
 });
 
-const baseCases = [
-  // chained members
-  {
-    code: 'foo && foo.bar',
-    output: 'foo?.bar',
-  },
-  {
-    code: 'foo.bar && foo.bar.baz',
-    output: 'foo.bar?.baz',
-  },
-  {
-    code: 'foo && foo()',
-    output: 'foo?.()',
-  },
-  {
-    code: 'foo.bar && foo.bar()',
-    output: 'foo.bar?.()',
-  },
-  {
-    code: 'foo && foo.bar && foo.bar.baz && foo.bar.baz.buzz',
-    output: 'foo?.bar?.baz?.buzz',
-  },
-  {
-    code: 'foo.bar && foo.bar.baz && foo.bar.baz.buzz',
-    output: 'foo.bar?.baz?.buzz',
-  },
-  // case with a jump (i.e. a non-nullish prop)
-  {
-    code: 'foo && foo.bar && foo.bar.baz.buzz',
-    output: 'foo?.bar?.baz.buzz',
-  },
-  {
-    code: 'foo.bar && foo.bar.baz.buzz',
-    output: 'foo.bar?.baz.buzz',
-  },
-  // case where for some reason there is a doubled up expression
-  {
-    code: 'foo && foo.bar && foo.bar.baz && foo.bar.baz && foo.bar.baz.buzz',
-    output: 'foo?.bar?.baz?.buzz',
-  },
-  {
-    code: 'foo.bar && foo.bar.baz && foo.bar.baz && foo.bar.baz.buzz',
-    output: 'foo.bar?.baz?.buzz',
-  },
-  // chained members with element access
-  {
-    code: 'foo && foo[bar] && foo[bar].baz && foo[bar].baz.buzz',
-    output: 'foo?.[bar]?.baz?.buzz',
-  },
-  {
+// eslint-disable-next-line @typescript-eslint/no-namespace
```

Comment:
@JoshuaKGoldberg I wanted to wrap the cases nicely in namespace, but the rule `@typescript-eslint/no-namespace` does not allow this. I tried to change to `module', but another rule `@typescript-eslint/prefer-namespace-keyword` throws an error.

Reply:
I would just drop the namespace - I don't think it adds much value.
if you want to isolate things, then I would move the test into a subfolder and isolate them in a separate module entirely. Eg

- `tests/rules/prefer-optional-chain/base-cases.ts`
- `tests/rules/prefer-optional-chain/prefer-optional-chain.test.ts`

Reply:
fixed